src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -327,15 +334,47 @@ internal PSVersionHashTable(IEqualityComparer equalityComparer) : base(equalityC
 
         /// <summary>
         /// Returns ordered collection with Keys of 'PSVersionHashTable'
+        /// We want see special order:
+        ///     1. PSVersionName
+        ///     2. PSEditionName
+        ///     3. Remaining properties in alphabetical order
         /// </summary>
         public override ICollection Keys
         {
             get
             {
-                Array arr = new string[base.Keys.Count];
-                base.Keys.CopyTo(arr, 0);
-                Array.Sort(arr, StringComparer.OrdinalIgnoreCase);
-                return arr;
+                ArrayList keyList = new ArrayList(base.Keys);
+                keyList.Sort(new PSVersionTableComparer());
+                return keyList;
+            }
+        }
+
+        private class PSVersionTableComparer : Comparer<object>
+        {
+            public override int Compare(object x, object y)
+            {
+                string xString = (string)LanguagePrimitives.ConvertTo(x, typeof(string), null);
+                string yString = (string)LanguagePrimitives.ConvertTo(y, typeof(string), null);
```

Comment:
This is very useful for me. 😄 The more we did not analyse this deeply before.

All previous options that we had, nothing broke really. The only thing we do here is to change a human output to the screen.
Output (Out-Default) always perform conversion to strings. Even if we have non-string key, it will be converted to a string. And it seems better to even see them sorted.

If a user adds a record with key that is converted to "PSVersion" we see two entries. We could block it but it will be breaking change.
Also we should use `LanguagePrimitives.ConvertTo<string>(Value)` (https://github.com/PowerShell/PowerShell/pull/2678#discussion_r87906295) to avoid "strange" sorting.

```powershell
$a=[pscustomobject]@{v1=123}
$a
$a.ToString() # It is empty!

$PSVersionTable[$a]="asdfgh"
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-alpha
PSEdition                      Core
@{v1=123}                      asdfgh
BuildVersion                   3.0.0.0
CLRVersion
GitCommitId                    v6.0.0-alpha.18-36-gc367cb58d87a84597077fbbe9a5ded5310179878
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0


$b=$a | Add-Member -MemberType ScriptMethod -Name ToString -Value {"PSVersion"} -PassThru -Force
$PSVersionTable[$b]="zxcvbn"
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      zxcvbn
PSVersion                      6.0.0-alpha
PSEdition                      Core
@{v1=123}                      asdfgh
BuildVersion                   3.0.0.0
CLRVersion
GitCommitId                    v6.0.0-alpha.18-36-gc367cb58d87a84597077fbbe9a5ded5310179878
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```